File: InMotionGIT.Common - Copy/MetadataCatalogAttributes/ElementLookUpAttribute.vb
Namespace Attributes

    ''' <summary>
    ''' Define como acceder la tabla que contiene la lista de valores hacer usado por la propiedad.
    ''' </summary>
    ''' <remarks></remarks>
    <AttributeUsage(AttributeTargets.Property)>
    Public NotInheritable Class ElementLookupAttribute
        Inherits Attribute

        Private _TableName As String
        Private _KeyField As String
        Private _DescriptionField As String
        Private _languageField As String

        Public ReadOnly Property TableName() As String
            Get
                Return _TableName
            End Get
        End Property

        Public ReadOnly Property KeyField() As String
            Get
                Return _KeyField
            End Get
        End Property

        Public ReadOnly Property DescriptionField() As String
            Get
                Return _DescriptionField
            End Get
        End Property

        Public ReadOnly Property LanguageField() As String
            Get
                Return _languageField
            End Get
        End Property

        Public Sub New(ByVal tableName As String, ByVal keyField As String, ByVal descriptionField As String)
            _TableName = tableName
            _KeyField = keyField
            _DescriptionField = descriptionField
        End Sub

        Public Sub New(ByVal tableName As String, ByVal keyField As String, ByVal descriptionField As String, ByVal languageField As String)
            _TableName = tableName
            _KeyField = keyField
            _DescriptionField = descriptionField
            _languageField = languageField
        End Sub

    End Class

End Namespace

File: Common/InMotionGIT.Common/MetadataCatalogAttributes/ElementLookUpAttribute.cs
using System;

namespace InMotionGIT.Common.Attributes
{

    /// <summary>
    /// Define como acceder la tabla que contiene la lista de valores hacer usado por la propiedad.
    /// </summary>
    /// <remarks></remarks>
    [AttributeUsage(AttributeTargets.Property)]
    public sealed class ElementLookupAttribute : Attribute
    {

        private string _TableName;
        private string _KeyField;
        private string _DescriptionField;
        private string _languageField;

        public string TableName
        {
            get
            {
                return _TableName;
            }
        }

        public string KeyField
        {
            get
            {
                return _KeyField;
            }
        }

        public string DescriptionField
        {
            get
            {
                return _DescriptionField;
            }
        }

        public string LanguageField
        {
            get
            {
                return _languageField;
            }
        }

        public ElementLookupAttribute(string tableName, string keyField, string descriptionField)
        {
            _TableName = tableName;
            _KeyField = keyField;
            _DescriptionField = descriptionField;
        }

        public ElementLookupAttribute(string tableName, string keyField, string descriptionField, string languageField)
        {
            _TableName = tableName;
            _KeyField = keyField;
            _DescriptionField = descriptionField;
            _languageField = languageField;
        }

    }

}